examples/mnist/train_mnist.py

```diff
@@ -86,8 +88,9 @@ def main():
     # The "main" refers to the target link of the "main" optimizer.
     trainer.extend(extensions.dump_graph('main/loss'))
 
-    # Take a snapshot at each epoch
-    trainer.extend(extensions.snapshot(), trigger=(args.epoch, 'epoch'))
+    # Take a snapshot for each specified epoch
+    frequency = args.frequency if args.frequency is not -1 else args.epoch
```

Comment:
Although it is a matter of tastes but I think `frequency = args.epoch if args.frequency == -1 else args.frequency` is better because negation could make the if statement more complicated.